components/dash-table/src/dash-table/dash/DataTable.js

```diff
@@ -253,27 +280,40 @@ export const propTypes = {
-             * `[false, ..., false, true]` and `[true, false, ..., false]`.
-             * If there are merged, multi-header columns then you can choose which column header
-             * row to display the input in by supplying an array of booleans.
-             * For example, `[true, false]` will display the `selectable` input on the first row,
-             * but now on the second row.
-             * If the `selectable` input appears on a merged columns, then clicking on that input
-             * will select *all* of the merged columns associated with it.
-             * The table-level prop `column_selectable` is used to determine the type of column
-             * selection to use.
+             * There are two `filter_options` props in the table.
+             * This is the column-level filter_options prop and there is
+             * also the table-level `filter_options` prop.
+             * These props determine whether the applicable filter relational
+             * operators will default to `sensitive` or `insensitive` comparison.
+             * If the column-level `filter_options` prop is set it overrides
+             * the table-level `filter_options` prop for that column.
+             */
+            filter_options: PropTypes.shape({
+                case: PropTypes.oneOf(['sensitive', 'insensitive'])
+            }),
+
+            /**
+             * The data-type provides support for per column typing and allows for data
+             * validation and coercion.
+             * 'numeric': represents both floats and ints.
+             * 'text': represents a string.
+             * 'datetime': a string representing a date or date-time, in the form:
+             *   'YYYY-MM-DD HH:MM:SS.ssssss' or some truncation thereof. Years must
+             *   have 4 digits, unless you use `validation.allow_YY: true`. Also
+             *   accepts 'T' or 't' between date and time, and allows timezone info
+             *   at the end. To convert these strings to Python `datetime` objects,
+             *   use `dateutil.parser.isoparse`. In R use `parse_iso_8601` from the
+             *   `parsedate` library.
+             *   WARNING: these parsers do not work with 2-digit years, if you use
+             *   `validation.allow_YY: true` and do not coerce to 4-digit years.
+             *   And parsers that do work with 2-digit years may make a different
+             *   guess about the century than we make on the front end.
+             * 'any': represents any type of data.
+             * Defaults to 'any' if undefined.
+             *
              *
              */
-            selectable: PropTypes.oneOfType([
-                PropTypes.oneOf(['first', 'last']),
-                PropTypes.bool,
-                PropTypes.arrayOf(PropTypes.bool)
-            ]),
+            type: PropTypes.oneOf(['any', 'numeric', 'text', 'datetime']),
```

Comment:
Do nested props also get their order preserved by this PR? If not don't worry about it for now, but assuming at some point they will, let's put `type` and `presentation` up top, right below `id` and `name`, as these are very important and commonly used.

Reply:
Updated in 15599ae.

Reply:
Unfortunately nested props aren't preserved. For the moment though, I've made the change in d9e386b.